br/pkg/restore/client_test.go

```diff
@@ -330,12 +331,35 @@ func TestPreCheckTableClusterIndex(t *testing.T) {
 type fakePDClient struct {
 	pd.Client
 	stores []*metapb.Store
+
+	notLeader bool
 }
 
 func (fpdc fakePDClient) GetAllStores(context.Context, ...pd.GetStoreOption) ([]*metapb.Store, error) {
 	return append([]*metapb.Store{}, fpdc.stores...), nil
 }
 
+func (fpdc fakePDClient) GetTS(ctx context.Context) (int64, int64, error) {
+	if fpdc.notLeader {
+		return 0, 0, errors.Errorf("rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster")
+	}
+	return 1, 1, nil
+}
+
+func TestGetTSWithRetry(t *testing.T) {
+	t.Log("case 1:")
+	pDClient := fakePDClient{notLeader: false}
+	client := restore.NewRestoreClient(pDClient, nil, defaultKeepaliveCfg, false)
+	_, err := client.GetTSWithRetry(context.Background())
+	require.NoError(t, err)
+
+	t.Log("case 2:")
+	pDClient = fakePDClient{notLeader: true}
+	client = restore.NewRestoreClient(pDClient, nil, defaultKeepaliveCfg, false)
+	_, err = client.GetTSWithRetry(context.Background())
+	require.Error(t, err)
```

Comment:
Can we add a case that the mock PD fails in the first request, then success at the second try? (Which should also return no errors.)

Reply:
I can simulate this case. But maybe this case is already covered when `get-ts-error` failpoint enable?

Reply:
I added a new unit test case to simulate this scenario. PTAL